superbench/benchmarks/micro_benchmarks/cudnn_test/cudnn_helper.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file cudnn_helper.cpp
+ * @brief  Cpp file for some functions related to cudnn
+ */
+
+#include <bits/stdc++.h>
+#include <stdlib.h>
+
+#include "cudnn_benchmark.h"
+
+namespace cudnn_test {
+/**
+ * @brief check cudnn function running status and throw error str
+ */
+void throw_cudnn_err(cudnnStatus_t result, const char *func, const char *file, int const line) {
+    if (result != CUDNN_STATUS_SUCCESS) {
+        const char *msg = cudnnGetErrorString(result);
+        std::stringstream safe_call_ss;
+        safe_call_ss << func << " failed with error"
+                     << "\nfile: " << file << "\nline: " << line << "\nmsg: " << msg;
+        throw std::runtime_error(safe_call_ss.str());
+    }
+}
+/**
+ * @brief check cudnn function running status and throw error str
+ */
+void check_cuda(cudaError_t result, const char *func, const char *file, int const line) {
+    if (result != cudaSuccess) {
+        const char *msg = cudaGetErrorString(result);
+        std::stringstream safe_call_ss;
+        safe_call_ss << func << " failed with error"
+                     << "\nfile: " << file << "\nline: " << line << "\nmsg: " << msg;
+        // Make sure we call CUDA Device Reset before exiting
+        throw std::runtime_error(safe_call_ss.str());
+    }
+}
+/**
+ * @brief Cuda context init
+ */
+void cuda_init(cudnnHandle_t *cudnn_handle) {
+    CUDA_SAFE_CALL(cudaDeviceReset());
+    CUDA_SAFE_CALL(cudaSetDevice(0));
```

Comment:
Only for one device?

Reply:
I think this code just test for one visible device, you can set cuda_visible_devices to run on other GPU

Reply:
I have tested that we can use cuda_visible_devices to decide which gpu to run
![image](https://user-images.githubusercontent.com/37182275/120147866-ef518480-c219-11eb-81b0-6708c188da30.png)

Reply:
what about `NVIDIA_VISIBLE_DEVICES`?

Reply:
what is NVIDIA_VISIBLE_DEVICES? It does not work.

Reply:
https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/user-guide.html#gpu-enumeration
it works for nvidia docker container I think，not for an application？